lib/helpers.js

```diff
@@ -0,0 +1,126 @@
+/*!
+ * Copyright (c) 2023 Digital Bazaar, Inc. All rights reserved.
+ */
+import {
+  Ed25519VerificationKey2020
+} from '@digitalbazaar/ed25519-verification-key-2020';
+import {
+  X25519KeyAgreementKey2020
+} from '@digitalbazaar/x25519-key-agreement-key-2020';
+import {
+  X25519KeyAgreementKey2019
+} from '@digitalbazaar/x25519-key-agreement-key-2019';
+
+const DID_CONTEXT_URL = 'https://www.w3.org/ns/did/v1';
+
+// For backwards compat only, not actually importing this suite
+const ED25519_KEY_2018_CONTEXT_URL =
+  'https://w3id.org/security/suites/ed25519-2018/v1';
+
+const contextsBySuite = new Map([
+  [Ed25519VerificationKey2020.suite, Ed25519VerificationKey2020.SUITE_CONTEXT],
+  ['Ed25519VerificationKey2018', ED25519_KEY_2018_CONTEXT_URL],
+  [X25519KeyAgreementKey2020.suite, X25519KeyAgreementKey2020.SUITE_CONTEXT],
+  [X25519KeyAgreementKey2019.suite, X25519KeyAgreementKey2019.SUITE_CONTEXT]
+]);
+
+/**
+ * Returns the public key object for a given key id fragment.
+ *
+ * @param {object} options - Options hashmap.
+ * @param {object} options.didDocument - The DID Document to use when generating
+ *   the id.
+ * @param {string} options.keyIdFragment - The key identifier fragment.
+ *
+ * @returns {object} Returns the public key node, with `@context`.
+ */
+export function _getKey({didDocument, keyIdFragment}) {
+  // Determine if the key id fragment belongs to the "main" public key,
+  // or the keyAgreement key
+  const keyId = didDocument.id + '#' + keyIdFragment;
+  let publicKey;
+
+  if(didDocument.verificationMethod[0].id === keyId) {
+    // Return the public key node for the main public key
+    publicKey = didDocument.verificationMethod[0];
+  } else {
+    // Return the public key node for the X25519 key-agreement key
+    publicKey = didDocument.keyAgreement[0];
+  }
+
+  return {
+    '@context': contextsBySuite.get(publicKey.type),
+    ...publicKey
+  };
+}
+
+export function _createVerificationSuite({generate, from, fromFingerprint}) {
+  return {
+    async getVerificationKeypair({...keyPair}) {
+      const verificationKeyPair = await from({...keyPair});
+
+      const did = verificationKeyPair.fingerprint ?
+        `did:key:${verificationKeyPair.fingerprint()}` :
+        `did:key:${verificationKeyPair.publicKeyMultibase}`;
+
+      verificationKeyPair.controller = did;
+      // Now set the source key's id
+      verificationKeyPair.id = verificationKeyPair.fingerprint ?
+        `${did}#${verificationKeyPair.fingerprint()}` :
+        `${did}#${verificationKeyPair.publicKeyMultibase}`;
+
+      return {verificationKeyPair, did};
+    },
+    async fromFingerprint({fingerprint}) {
+      // Ed25519VerificationKey2018 from expects `publicKeyBase58` param
+      // instead of `publicKeyMultibase`.
```

Comment:
NOTE:  Ed25519VerificationKey2018 `from()` expects `publicKeyBase58` param instead of `publicKeyMultibase` and when I pass fingerprint to `publicKeyBase58`, I get the following error 

```    
Error: Expected 32 bytes
      at ensureBytes (file:///Users/tashid.gyeltshen/Desktop/DigitalBazaar/did-method-key/node_modules/@noble/ed25519/lib/esm/index.js:621:15)
      at Function.fromHex (file:///Users/tashid.gyeltshen/Desktop/DigitalBazaar/did-method-key/node_modules/@noble/ed25519/lib/esm/index.js:363:15)
      at Function.convertFromEdPublicKey (file:///Users/tashid.gyeltshen/Desktop/DigitalBazaar/did-method-key/node_modules/@digitalbazaar/x25519-key-agreement-key-2019/lib/X25519KeyAgreementKey2019.js:212:33)
      at Function.fromEd25519VerificationKey2018 (file:///Users/tashid.gyeltshen/Desktop/DigitalBazaar/did-method-key/node_modules/@digitalbazaar/x25519-key-agreement-key-2019/lib/X25519KeyAgreementKey2019.js:123:10)
      at _getKeyAgreementKeyPair (file:///Users/tashid.gyeltshen/Desktop/DigitalBazaar/did-method-key/lib/helpers.js:99:10)
      at DidKeyDriver._keyPairToDidDocument (file:///Users/tashid.gyeltshen/Desktop/DigitalBazaar/did-method-key/lib/DidKeyDriver.js:178:51)
      at async DidKeyDriver.get (file:///Users/tashid.gyeltshen/Desktop/DigitalBazaar/did-method-key/lib/DidKeyDriver.js:130:27)
      at async Context.<anonymous> (file:///Users/tashid.gyeltshen/Desktop/DigitalBazaar/did-method-key/test/driver.spec.js:187:19)
````

Reply:
Note that the adapter should not be used on non-multikey generate/from methods. The `createVerificationSuite` helper should be exposed to the caller and they should only pass in `generate` and `from` methods that match the requirements for adaption.